### ISSUE TITLE
Allow kscript cache dir to be configurable

### ISSUE THREAD

**Issue by seanabraham** [holgerbrandl/kscript#146]
It would be really nice if we could configure where we put the kscript cache dir. My use case is a bit odd  — doing an initial kscript invocation with one user and then wanting to use the cached results from another user, but I still do think that configurability could be valuable to others here.

**Comment by seanabraham** [comment]
If you're open to PRs, I could try to put one together

**Comment by holgerbrandl** [comment]
What about using an environment variable (like `KSCRIPT_CACHE_DIR`? It typically is less invasive. Sure PRs are welcome.

**Comment by felipecsl** [comment]
Another quirk with dependencies that I just noticed is that if you delete any previously downloaded dependencies from ~/.m2, future kscript invocations for the same script will fail with 
```
[kscript] [ERROR] compilation of './CITest.kts' failed
warning: classpath entry points to a non-existent location: /Users/jonh_doe/.m2/repository/deleted/dependency...
```

kscript will no longer work until you manually delete `~/.kscript` and rerun it, which sounds counter intuitive. Should I open a separate issue for this?

**Comment by holgerbrandl** [comment]
Indeed, that's a separate issue. But it should be easy to fix, since we "just" have to make sure that the cached classpath elements exist, and rerun maven if not.
